// @pulumi/gcp/biglake/icebergNamespace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IcebergNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * IcebergNamespaces are containers for Apache Iceberg Tables within an IcebergCatalog.
 *
 * ## Example Usage
 *
 * ### Biglake Iceberg Namespace
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "example-bucket",
 *     location: "us-central1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 * });
 * const catalog = new gcp.biglake.IcebergCatalog("catalog", {
 *     name: bucket.name,
 *     catalogType: "CATALOG_TYPE_GCS_BUCKET",
 * });
 * const myIcebergNamespace = new gcp.biglake.IcebergNamespace("my_iceberg_namespace", {
 *     catalog: catalog.name,
 *     namespaceId: "my_iceberg_namespace",
 *     properties: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * IcebergNamespace can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/catalogs/{{catalog}}/namespaces/{{namespace_id}}`
 * * `{{project}}/{{catalog}}/{{namespace_id}}`
 * * `{{catalog}}/{{namespace_id}}`
 *
 * When using the `pulumi import` command, IcebergNamespace can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:biglake/icebergNamespace:IcebergNamespace default projects/{{project}}/catalogs/{{catalog}}/namespaces/{{namespace_id}}
 * $ pulumi import gcp:biglake/icebergNamespace:IcebergNamespace default {{project}}/{{catalog}}/{{namespace_id}}
 * $ pulumi import gcp:biglake/icebergNamespace:IcebergNamespace default {{catalog}}/{{namespace_id}}
 * ```
 */
class IcebergNamespace extends pulumi.CustomResource {
    /**
     * Get an existing IcebergNamespace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IcebergNamespace(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of IcebergNamespace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IcebergNamespace.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["catalog"] = state?.catalog;
            resourceInputs["namespaceId"] = state?.namespaceId;
            resourceInputs["project"] = state?.project;
            resourceInputs["properties"] = state?.properties;
        }
        else {
            const args = argsOrState;
            if (args?.catalog === undefined && !opts.urn) {
                throw new Error("Missing required property 'catalog'");
            }
            if (args?.namespaceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'namespaceId'");
            }
            resourceInputs["catalog"] = args?.catalog;
            resourceInputs["namespaceId"] = args?.namespaceId;
            resourceInputs["project"] = args?.project;
            resourceInputs["properties"] = args?.properties;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IcebergNamespace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IcebergNamespace = IcebergNamespace;
/** @internal */
IcebergNamespace.__pulumiType = 'gcp:biglake/icebergNamespace:IcebergNamespace';
