// @pulumi/gcp/vectorsearch/collection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Collection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Description
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * ## Example Usage
 *
 * ### Vectorsearch Collection Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example_collection = new gcp.vectorsearch.Collection("example-collection", {
 *     location: "us-central1",
 *     collectionId: "example-collection",
 *     displayName: "My Awesome Collection",
 *     description: "This collection stores important data.",
 *     labels: {
 *         env: "dev",
 *         team: "my-team",
 *     },
 *     dataSchema: `{
 *   \\"type\\": \\"object\\",
 *   \\"properties\\": {
 *     \\"title\\": {
 *       \\"type\\": \\"string\\"
 *     },
 *     \\"plot\\": {
 *       \\"type\\": \\"string\\"
 *     }
 *   }
 * }
 * `,
 *     vectorSchemas: [{
 *         fieldName: "text_embedding",
 *         denseVector: {
 *             dimensions: 768,
 *             vertexEmbeddingConfig: {
 *                 modelId: "textembedding-gecko@003",
 *                 taskType: "RETRIEVAL_DOCUMENT",
 *                 textTemplate: "Title: {title} ---- Plot: {plot}",
 *             },
 *         },
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Collection can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/collections/{{collection_id}}`
 * * `{{project}}/{{location}}/{{collection_id}}`
 * * `{{location}}/{{collection_id}}`
 *
 * When using the `pulumi import` command, Collection can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:vectorsearch/collection:Collection default projects/{{project}}/locations/{{location}}/collections/{{collection_id}}
 * $ pulumi import gcp:vectorsearch/collection:Collection default {{project}}/{{location}}/{{collection_id}}
 * $ pulumi import gcp:vectorsearch/collection:Collection default {{location}}/{{collection_id}}
 * ```
 */
class Collection extends pulumi.CustomResource {
    /**
     * Get an existing Collection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Collection(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Collection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Collection.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["collectionId"] = state?.collectionId;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["dataSchema"] = state?.dataSchema;
            resourceInputs["description"] = state?.description;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["vectorSchemas"] = state?.vectorSchemas;
        }
        else {
            const args = argsOrState;
            if (args?.collectionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'collectionId'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["collectionId"] = args?.collectionId;
            resourceInputs["dataSchema"] = args?.dataSchema;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["vectorSchemas"] = args?.vectorSchemas;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Collection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Collection = Collection;
/** @internal */
Collection.__pulumiType = 'gcp:vectorsearch/collection:Collection';
