// @pulumi/gcp/ces/deployment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Deployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Description
 *
 * ## Example Usage
 *
 * ### Ces Deployment Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const my_app = new gcp.ces.App("my-app", {
 *     location: "us",
 *     displayName: "my-app",
 *     appId: "app-id",
 *     timeZoneSettings: {
 *         timeZone: "America/Los_Angeles",
 *     },
 * });
 * const my_deployment = new gcp.ces.Deployment("my-deployment", {
 *     location: "us",
 *     displayName: "my-deployment",
 *     app: my_app.name,
 *     appVersion: "projects/example-project/locations/us/apps/example-app/versions/example-version",
 *     channelProfile: {
 *         channelType: "API",
 *         disableBargeInControl: true,
 *         disableDtmf: true,
 *         personaProperty: {
 *             persona: "CHATTY",
 *         },
 *         profileId: "temp_profile_id",
 *         webWidgetConfig: {
 *             modality: "CHAT_AND_VOICE",
 *             theme: "DARK",
 *             webWidgetTitle: "temp_webwidget_title",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Deployment can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/apps/{{app}}/deployments/{{name}}`
 * * `{{project}}/{{location}}/{{app}}/{{name}}`
 * * `{{location}}/{{app}}/{{name}}`
 *
 * When using the `pulumi import` command, Deployment can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:ces/deployment:Deployment default projects/{{project}}/locations/{{location}}/apps/{{app}}/deployments/{{name}}
 * $ pulumi import gcp:ces/deployment:Deployment default {{project}}/{{location}}/{{app}}/{{name}}
 * $ pulumi import gcp:ces/deployment:Deployment default {{location}}/{{app}}/{{name}}
 * ```
 */
class Deployment extends pulumi.CustomResource {
    /**
     * Get an existing Deployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Deployment(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Deployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Deployment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["app"] = state?.app;
            resourceInputs["appVersion"] = state?.appVersion;
            resourceInputs["channelProfile"] = state?.channelProfile;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.app === undefined && !opts.urn) {
                throw new Error("Missing required property 'app'");
            }
            if (args?.appVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'appVersion'");
            }
            if (args?.channelProfile === undefined && !opts.urn) {
                throw new Error("Missing required property 'channelProfile'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["app"] = args?.app;
            resourceInputs["appVersion"] = args?.appVersion;
            resourceInputs["channelProfile"] = args?.channelProfile;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Deployment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Deployment = Deployment;
/** @internal */
Deployment.__pulumiType = 'gcp:ces/deployment:Deployment';
