// @pulumi/gcp/ces/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Toolset = exports.Tool = exports.Guardrail = exports.Example = exports.Deployment = exports.AppVersion = exports.App = exports.Agent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Agent = null;
utilities.lazyLoad(exports, ["Agent"], () => require("./agent"));
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.AppVersion = null;
utilities.lazyLoad(exports, ["AppVersion"], () => require("./appVersion"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.Example = null;
utilities.lazyLoad(exports, ["Example"], () => require("./example"));
exports.Guardrail = null;
utilities.lazyLoad(exports, ["Guardrail"], () => require("./guardrail"));
exports.Tool = null;
utilities.lazyLoad(exports, ["Tool"], () => require("./tool"));
exports.Toolset = null;
utilities.lazyLoad(exports, ["Toolset"], () => require("./toolset"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:ces/agent:Agent":
                return new exports.Agent(name, undefined, { urn });
            case "gcp:ces/app:App":
                return new exports.App(name, undefined, { urn });
            case "gcp:ces/appVersion:AppVersion":
                return new exports.AppVersion(name, undefined, { urn });
            case "gcp:ces/deployment:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "gcp:ces/example:Example":
                return new exports.Example(name, undefined, { urn });
            case "gcp:ces/guardrail:Guardrail":
                return new exports.Guardrail(name, undefined, { urn });
            case "gcp:ces/tool:Tool":
                return new exports.Tool(name, undefined, { urn });
            case "gcp:ces/toolset:Toolset":
                return new exports.Toolset(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "ces/agent", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/app", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/appVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/deployment", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/example", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/guardrail", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/tool", _module);
pulumi.runtime.registerResourceModule("gcp", "ces/toolset", _module);
