// @pulumi/gcp/tpu/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Vm = exports.V2QueuedResource = exports.getV2RuntimeVersionsOutput = exports.getV2RuntimeVersions = exports.getV2AcceleratorTypesOutput = exports.getV2AcceleratorTypes = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getV2AcceleratorTypes = null;
exports.getV2AcceleratorTypesOutput = null;
utilities.lazyLoad(exports, ["getV2AcceleratorTypes", "getV2AcceleratorTypesOutput"], () => require("./getV2AcceleratorTypes"));
exports.getV2RuntimeVersions = null;
exports.getV2RuntimeVersionsOutput = null;
utilities.lazyLoad(exports, ["getV2RuntimeVersions", "getV2RuntimeVersionsOutput"], () => require("./getV2RuntimeVersions"));
exports.V2QueuedResource = null;
utilities.lazyLoad(exports, ["V2QueuedResource"], () => require("./v2queuedResource"));
exports.V2Vm = null;
utilities.lazyLoad(exports, ["V2Vm"], () => require("./v2vm"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:tpu/v2QueuedResource:V2QueuedResource":
                return new exports.V2QueuedResource(name, undefined, { urn });
            case "gcp:tpu/v2Vm:V2Vm":
                return new exports.V2Vm(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "tpu/v2QueuedResource", _module);
pulumi.runtime.registerResourceModule("gcp", "tpu/v2Vm", _module);
