// @pulumi/gcp/dataproc/autoscalingPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoscalingPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes an autoscaling policy for Dataproc cluster autoscaler.
 *
 * ## Example Usage
 *
 * ### Dataproc Autoscaling Policy
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const asp = new gcp.dataproc.AutoscalingPolicy("asp", {
 *     policyId: "dataproc-policy",
 *     location: "us-central1",
 *     workerConfig: {
 *         maxInstances: 3,
 *     },
 *     basicAlgorithm: {
 *         yarnConfig: {
 *             gracefulDecommissionTimeout: "30s",
 *             scaleUpFactor: 0.5,
 *             scaleDownFactor: 0.5,
 *         },
 *     },
 * });
 * const basic = new gcp.dataproc.Cluster("basic", {
 *     name: "dataproc-policy",
 *     region: "us-central1",
 *     clusterConfig: {
 *         autoscalingConfig: {
 *             policyUri: asp.name,
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * AutoscalingPolicy can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/autoscalingPolicies/{{policy_id}}`
 * * `{{project}}/{{location}}/{{policy_id}}`
 * * `{{location}}/{{policy_id}}`
 *
 * When using the `pulumi import` command, AutoscalingPolicy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataproc/autoscalingPolicy:AutoscalingPolicy default projects/{{project}}/locations/{{location}}/autoscalingPolicies/{{policy_id}}
 * $ pulumi import gcp:dataproc/autoscalingPolicy:AutoscalingPolicy default {{project}}/{{location}}/{{policy_id}}
 * $ pulumi import gcp:dataproc/autoscalingPolicy:AutoscalingPolicy default {{location}}/{{policy_id}}
 * ```
 */
class AutoscalingPolicy extends pulumi.CustomResource {
    /**
     * Get an existing AutoscalingPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AutoscalingPolicy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AutoscalingPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AutoscalingPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["basicAlgorithm"] = state?.basicAlgorithm;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["policyId"] = state?.policyId;
            resourceInputs["project"] = state?.project;
            resourceInputs["secondaryWorkerConfig"] = state?.secondaryWorkerConfig;
            resourceInputs["workerConfig"] = state?.workerConfig;
        }
        else {
            const args = argsOrState;
            if (args?.policyId === undefined && !opts.urn) {
                throw new Error("Missing required property 'policyId'");
            }
            resourceInputs["basicAlgorithm"] = args?.basicAlgorithm;
            resourceInputs["location"] = args?.location;
            resourceInputs["policyId"] = args?.policyId;
            resourceInputs["project"] = args?.project;
            resourceInputs["secondaryWorkerConfig"] = args?.secondaryWorkerConfig;
            resourceInputs["workerConfig"] = args?.workerConfig;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AutoscalingPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AutoscalingPolicy = AutoscalingPolicy;
/** @internal */
AutoscalingPolicy.__pulumiType = 'gcp:dataproc/autoscalingPolicy:AutoscalingPolicy';
