// @pulumi/gcp/identityplatform/defaultSupportedIdpConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultSupportedIdpConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Configurations options for authenticating with a the standard set of Identity Toolkit-trusted IDPs.
 *
 * You must enable the
 * [Google Identity Platform](https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity) in
 * the marketplace prior to using this resource.
 *
 * ## Example Usage
 *
 * ### Identity Platform Default Supported Idp Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const idpConfig = new gcp.identityplatform.DefaultSupportedIdpConfig("idp_config", {
 *     enabled: true,
 *     idpId: "playgames.google.com",
 *     clientId: "client-id",
 *     clientSecret: "secret",
 * });
 * ```
 *
 * ## Import
 *
 * DefaultSupportedIdpConfig can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/defaultSupportedIdpConfigs/{{idp_id}}`
 * * `{{project}}/{{idp_id}}`
 * * `{{idp_id}}`
 *
 * When using the `pulumi import` command, DefaultSupportedIdpConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:identityplatform/defaultSupportedIdpConfig:DefaultSupportedIdpConfig default projects/{{project}}/defaultSupportedIdpConfigs/{{idp_id}}
 * $ pulumi import gcp:identityplatform/defaultSupportedIdpConfig:DefaultSupportedIdpConfig default {{project}}/{{idp_id}}
 * $ pulumi import gcp:identityplatform/defaultSupportedIdpConfig:DefaultSupportedIdpConfig default {{idp_id}}
 * ```
 */
class DefaultSupportedIdpConfig extends pulumi.CustomResource {
    /**
     * Get an existing DefaultSupportedIdpConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DefaultSupportedIdpConfig(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DefaultSupportedIdpConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DefaultSupportedIdpConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["clientId"] = state?.clientId;
            resourceInputs["clientSecret"] = state?.clientSecret;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["idpId"] = state?.idpId;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
        }
        else {
            const args = argsOrState;
            if (args?.clientId === undefined && !opts.urn) {
                throw new Error("Missing required property 'clientId'");
            }
            if (args?.clientSecret === undefined && !opts.urn) {
                throw new Error("Missing required property 'clientSecret'");
            }
            if (args?.idpId === undefined && !opts.urn) {
                throw new Error("Missing required property 'idpId'");
            }
            resourceInputs["clientId"] = args?.clientId;
            resourceInputs["clientSecret"] = args?.clientSecret;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["idpId"] = args?.idpId;
            resourceInputs["project"] = args?.project;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DefaultSupportedIdpConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DefaultSupportedIdpConfig = DefaultSupportedIdpConfig;
/** @internal */
DefaultSupportedIdpConfig.__pulumiType = 'gcp:identityplatform/defaultSupportedIdpConfig:DefaultSupportedIdpConfig';
