// @pulumi/gcp/networkservices/mesh.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mesh = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Mesh represents a logical configuration grouping for workload to workload communication within a
 * service mesh. Routes that point to mesh dictate how requests are routed within this logical
 * mesh boundary.
 *
 * To get more information about Mesh, see:
 *
 * * [API documentation](https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.meshes)
 *
 * ## Example Usage
 *
 * ### Network Services Mesh Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.Mesh("default", {
 *     name: "my-mesh",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     interceptionPort: 443,
 * });
 * ```
 * ### Network Services Mesh No Port
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.Mesh("default", {
 *     name: "my-mesh-noport",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 * });
 * ```
 * ### Network Services Mesh Location
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.Mesh("default", {
 *     name: "my-mesh",
 *     location: "global",
 * });
 * ```
 *
 * ## Import
 *
 * Mesh can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/meshes/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, Mesh can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/mesh:Mesh default projects/{{project}}/locations/{{location}}/meshes/{{name}}
 * $ pulumi import gcp:networkservices/mesh:Mesh default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:networkservices/mesh:Mesh default {{location}}/{{name}}
 * ```
 */
class Mesh extends pulumi.CustomResource {
    /**
     * Get an existing Mesh resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Mesh(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Mesh.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Mesh.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["interceptionPort"] = state?.interceptionPort;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["selfLink"] = state?.selfLink;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            resourceInputs["description"] = args?.description;
            resourceInputs["interceptionPort"] = args?.interceptionPort;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Mesh.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Mesh = Mesh;
/** @internal */
Mesh.__pulumiType = 'gcp:networkservices/mesh:Mesh';
