// @pulumi/gcp/apigee/api.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * To get more information about API proxies see, see:
 *
 * * [API documentation](https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.apis)
 * * How-to Guides
 *   * [API proxies](https://cloud.google.com/apigee/docs/resources)
 *
 * ## Import
 *
 * An API proxy can be imported using any of these accepted formats:
 *
 * * `{{org_id}}/apis/{{name}}`
 * * `{{org_id}}/{{name}}`
 *
 * When using the `pulumi import` command, API proxy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:apigee/api:Api default {{org_id}}/apis/{{name}}
 * $ pulumi import gcp:apigee/api:Api default {{org_id}}/{{name}}
 * ```
 */
class Api extends pulumi.CustomResource {
    /**
     * Get an existing Api resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Api(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Api.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Api.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["configBundle"] = state?.configBundle;
            resourceInputs["detectMd5hash"] = state?.detectMd5hash;
            resourceInputs["latestRevisionId"] = state?.latestRevisionId;
            resourceInputs["md5hash"] = state?.md5hash;
            resourceInputs["metaDatas"] = state?.metaDatas;
            resourceInputs["name"] = state?.name;
            resourceInputs["orgId"] = state?.orgId;
            resourceInputs["revisions"] = state?.revisions;
        }
        else {
            const args = argsOrState;
            if (args?.configBundle === undefined && !opts.urn) {
                throw new Error("Missing required property 'configBundle'");
            }
            if (args?.orgId === undefined && !opts.urn) {
                throw new Error("Missing required property 'orgId'");
            }
            resourceInputs["configBundle"] = args?.configBundle;
            resourceInputs["detectMd5hash"] = args?.detectMd5hash;
            resourceInputs["name"] = args?.name;
            resourceInputs["orgId"] = args?.orgId;
            resourceInputs["latestRevisionId"] = undefined /*out*/;
            resourceInputs["md5hash"] = undefined /*out*/;
            resourceInputs["metaDatas"] = undefined /*out*/;
            resourceInputs["revisions"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Api.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Api = Api;
/** @internal */
Api.__pulumiType = 'gcp:apigee/api:Api';
