// @pulumi/gcp/discoveryengine/aclConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AclConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Access Control Configuration.
 *
 * To get more information about AclConfig, see:
 *
 * * [API documentation](https://cloud.google.com/generative-ai-app-builder/docs/reference/rpc/google.cloud.discoveryengine.v1alpha#aclconfigservice)
 *
 * ## Example Usage
 *
 * ### Discoveryengine Aclconfig Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const basic = new gcp.discoveryengine.AclConfig("basic", {
 *     location: "global",
 *     idpConfig: {
 *         idpType: "THIRD_PARTY",
 *         externalIdpConfig: {
 *             workforcePoolName: "locations/global/workforcePools/cloud-console-pool-manual",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * AclConfig can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/aclConfig`
 * * `{{project}}/{{location}}`
 * * `{{location}}`
 *
 * When using the `pulumi import` command, AclConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:discoveryengine/aclConfig:AclConfig default projects/{{project}}/locations/{{location}}/aclConfig
 * $ pulumi import gcp:discoveryengine/aclConfig:AclConfig default {{project}}/{{location}}
 * $ pulumi import gcp:discoveryengine/aclConfig:AclConfig default {{location}}
 * ```
 */
class AclConfig extends pulumi.CustomResource {
    /**
     * Get an existing AclConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AclConfig(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AclConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AclConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["idpConfig"] = state?.idpConfig;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["idpConfig"] = args?.idpConfig;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AclConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AclConfig = AclConfig;
/** @internal */
AclConfig.__pulumiType = 'gcp:discoveryengine/aclConfig:AclConfig';
