// @pulumi/gcp/discoveryengine/licenseConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * License Configuration.
 *
 * To get more information about LicenseConfig, see:
 *
 * * [API documentation](https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.licenseConfigs)
 *
 * ## Example Usage
 *
 * ### Discoveryengine Licenseconfig Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const basic = new gcp.discoveryengine.LicenseConfig("basic", {
 *     location: "global",
 *     licenseConfigId: "license-config-id",
 *     licenseCount: 50,
 *     subscriptionTier: "SUBSCRIPTION_TIER_SEARCH_AND_ASSISTANT",
 *     startDate: {
 *         year: 2099,
 *         month: 1,
 *         day: 1,
 *     },
 *     endDate: {
 *         year: 2100,
 *         month: 1,
 *         day: 1,
 *     },
 *     subscriptionTerm: "SUBSCRIPTION_TERM_ONE_YEAR",
 * });
 * ```
 *
 * ## Import
 *
 * LicenseConfig can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/licenseConfigs/{{license_config_id}}`
 * * `{{project}}/{{location}}/{{license_config_id}}`
 * * `{{location}}/{{license_config_id}}`
 *
 * When using the `pulumi import` command, LicenseConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:discoveryengine/licenseConfig:LicenseConfig default projects/{{project}}/locations/{{location}}/licenseConfigs/{{license_config_id}}
 * $ pulumi import gcp:discoveryengine/licenseConfig:LicenseConfig default {{project}}/{{location}}/{{license_config_id}}
 * $ pulumi import gcp:discoveryengine/licenseConfig:LicenseConfig default {{location}}/{{license_config_id}}
 * ```
 */
class LicenseConfig extends pulumi.CustomResource {
    /**
     * Get an existing LicenseConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LicenseConfig(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of LicenseConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LicenseConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["autoRenew"] = state?.autoRenew;
            resourceInputs["endDate"] = state?.endDate;
            resourceInputs["freeTrial"] = state?.freeTrial;
            resourceInputs["licenseConfigId"] = state?.licenseConfigId;
            resourceInputs["licenseCount"] = state?.licenseCount;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["startDate"] = state?.startDate;
            resourceInputs["subscriptionTerm"] = state?.subscriptionTerm;
            resourceInputs["subscriptionTier"] = state?.subscriptionTier;
        }
        else {
            const args = argsOrState;
            if (args?.licenseConfigId === undefined && !opts.urn) {
                throw new Error("Missing required property 'licenseConfigId'");
            }
            if (args?.licenseCount === undefined && !opts.urn) {
                throw new Error("Missing required property 'licenseCount'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.startDate === undefined && !opts.urn) {
                throw new Error("Missing required property 'startDate'");
            }
            if (args?.subscriptionTerm === undefined && !opts.urn) {
                throw new Error("Missing required property 'subscriptionTerm'");
            }
            if (args?.subscriptionTier === undefined && !opts.urn) {
                throw new Error("Missing required property 'subscriptionTier'");
            }
            resourceInputs["autoRenew"] = args?.autoRenew;
            resourceInputs["endDate"] = args?.endDate;
            resourceInputs["freeTrial"] = args?.freeTrial;
            resourceInputs["licenseConfigId"] = args?.licenseConfigId;
            resourceInputs["licenseCount"] = args?.licenseCount;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["startDate"] = args?.startDate;
            resourceInputs["subscriptionTerm"] = args?.subscriptionTerm;
            resourceInputs["subscriptionTier"] = args?.subscriptionTier;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LicenseConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LicenseConfig = LicenseConfig;
/** @internal */
LicenseConfig.__pulumiType = 'gcp:discoveryengine/licenseConfig:LicenseConfig';
