// @pulumi/gcp/compute/regionHealthAggregationPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionHealthAggregationPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Health Aggregation Policy specifies how to aggregate the service's constituent source health status to determine an aggregated health status result for the service.  For example, you can specify a criteria such as “Consider the Health Source as `Healthy` if there are at least 5 healthy backend endpoints”. Or, you can specify a criteria like “Consider the Health Source as `Healthy` if minimum 60% of endpoints in the Backend are healthy”. The Health Aggregation Policy is applied to each Health Source (e.g. Backend Service) individually.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about RegionHealthAggregationPolicy, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/rest/v1/healthAggregationPolicies)
 * * How-to Guides
 *     * [Health checks overview](https://cloud.google.com/load-balancing/docs/health-check-concepts)
 *
 * ## Example Usage
 *
 * ### Compute Region Health Aggregation Policy Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const exampleTestHealthAggregationPolicy = new gcp.compute.RegionHealthAggregationPolicy("example_test_health_aggregation_policy", {
 *     name: "test-health-aggregation-policy",
 *     description: "Example health aggregation policy basic",
 *     region: "us-central1",
 * });
 * ```
 *
 * ## Import
 *
 * RegionHealthAggregationPolicy can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/regions/{{region}}/healthAggregationPolicies/{{name}}`
 * * `{{project}}/{{region}}/{{name}}`
 * * `{{region}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, RegionHealthAggregationPolicy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy default projects/{{project}}/regions/{{region}}/healthAggregationPolicies/{{name}}
 * $ pulumi import gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy default {{project}}/{{region}}/{{name}}
 * $ pulumi import gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy default {{region}}/{{name}}
 * $ pulumi import gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy default {{name}}
 * ```
 */
class RegionHealthAggregationPolicy extends pulumi.CustomResource {
    /**
     * Get an existing RegionHealthAggregationPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegionHealthAggregationPolicy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RegionHealthAggregationPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegionHealthAggregationPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["creationTimestamp"] = state?.creationTimestamp;
            resourceInputs["description"] = state?.description;
            resourceInputs["fingerprint"] = state?.fingerprint;
            resourceInputs["healthyPercentThreshold"] = state?.healthyPercentThreshold;
            resourceInputs["minHealthyThreshold"] = state?.minHealthyThreshold;
            resourceInputs["name"] = state?.name;
            resourceInputs["policyType"] = state?.policyType;
            resourceInputs["project"] = state?.project;
            resourceInputs["region"] = state?.region;
            resourceInputs["selfLinkWithId"] = state?.selfLinkWithId;
        }
        else {
            const args = argsOrState;
            if (args?.region === undefined && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["healthyPercentThreshold"] = args?.healthyPercentThreshold;
            resourceInputs["minHealthyThreshold"] = args?.minHealthyThreshold;
            resourceInputs["name"] = args?.name;
            resourceInputs["policyType"] = args?.policyType;
            resourceInputs["project"] = args?.project;
            resourceInputs["region"] = args?.region;
            resourceInputs["creationTimestamp"] = undefined /*out*/;
            resourceInputs["fingerprint"] = undefined /*out*/;
            resourceInputs["selfLinkWithId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegionHealthAggregationPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegionHealthAggregationPolicy = RegionHealthAggregationPolicy;
/** @internal */
RegionHealthAggregationPolicy.__pulumiType = 'gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy';
