// @pulumi/gcp/apihub/apiHubInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiHubInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An ApiHubInstance represents the instance resources of the API Hub.
 * Currently, only one ApiHub instance is allowed for each project.
 * Currently, updation/deletion of ApiHub instance is not allowed.
 *
 * ## Example Usage
 *
 * ### Apihub Api Hub Instance Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const apihub_instance_without_search = new gcp.apihub.ApiHubInstance("apihub-instance-without-search", {
 *     location: "us-central1",
 *     config: {
 *         disableSearch: true,
 *     },
 * });
 * ```
 * ### Apihub Api Hub Instance Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const apihub_instance_search = new gcp.apihub.ApiHubInstance("apihub-instance-search", {
 *     project: "my-project",
 *     apiHubInstanceId: "test-instance-full",
 *     description: "Test API hub instance",
 *     location: "us-central1",
 *     config: {
 *         encryptionType: "CMEK",
 *         cmekKeyName: "projects/my-project/locations/us-central1/keyRings/apihub/cryptoKeys/apihub-key",
 *         disableSearch: false,
 *         vertexLocation: "us",
 *     },
 *     labels: {
 *         environment: "dev",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ApiHubInstance can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/apiHubInstances/{{api_hub_instance_id}}`
 * * `{{project}}/{{location}}/{{api_hub_instance_id}}`
 * * `{{location}}/{{api_hub_instance_id}}`
 *
 * When using the `pulumi import` command, ApiHubInstance can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:apihub/apiHubInstance:ApiHubInstance default projects/{{project}}/locations/{{location}}/apiHubInstances/{{api_hub_instance_id}}
 * $ pulumi import gcp:apihub/apiHubInstance:ApiHubInstance default {{project}}/{{location}}/{{api_hub_instance_id}}
 * $ pulumi import gcp:apihub/apiHubInstance:ApiHubInstance default {{location}}/{{api_hub_instance_id}}
 * ```
 */
class ApiHubInstance extends pulumi.CustomResource {
    /**
     * Get an existing ApiHubInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ApiHubInstance(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ApiHubInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApiHubInstance.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["apiHubInstanceId"] = state?.apiHubInstanceId;
            resourceInputs["config"] = state?.config;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["state"] = state?.state;
            resourceInputs["stateMessage"] = state?.stateMessage;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.config === undefined && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["apiHubInstanceId"] = args?.apiHubInstanceId;
            resourceInputs["config"] = args?.config;
            resourceInputs["description"] = args?.description;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["stateMessage"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(ApiHubInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ApiHubInstance = ApiHubInstance;
/** @internal */
ApiHubInstance.__pulumiType = 'gcp:apihub/apiHubInstance:ApiHubInstance';
