// @pulumi/gcp/saasruntime/rolloutKind.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RolloutKind = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A RolloutKind is a reusable configuration resource that defines the policies, strategies, and targeting for Rollout operations. It acts as a template for repeatable Rollouts, providing guardrails and ensuring that updates are executed in a consistent manner across a fleet of Units.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * ## Example Usage
 *
 * ### Saas Runtime Rollout Kind Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const exampleSaas = new gcp.saasruntime.SaaS("example_saas", {
 *     saasId: "example-saas",
 *     location: "global",
 *     locations: [{
 *         name: "us-central1",
 *     }],
 * });
 * const exampleUnitkind = new gcp.saasruntime.UnitKind("example_unitkind", {
 *     location: "global",
 *     unitKindId: "example-unitkind",
 *     saas: exampleSaas.id,
 * });
 * const example = new gcp.saasruntime.RolloutKind("example", {
 *     location: "global",
 *     rolloutKindId: "example-rolloutkind",
 *     unitKind: exampleUnitkind.id,
 *     rolloutOrchestrationStrategy: "Google.Cloud.Simple.OneLocationAtATime",
 *     errorBudget: {
 *         allowedCount: 1,
 *     },
 *     unitFilter: "unit.labels['key1'] == 'value1'",
 * });
 * ```
 *
 * ## Import
 *
 * RolloutKind can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/rolloutKinds/{{rollout_kind_id}}`
 * * `{{project}}/{{location}}/{{rollout_kind_id}}`
 * * `{{location}}/{{rollout_kind_id}}`
 *
 * When using the `pulumi import` command, RolloutKind can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:saasruntime/rolloutKind:RolloutKind default projects/{{project}}/locations/{{location}}/rolloutKinds/{{rollout_kind_id}}
 * $ pulumi import gcp:saasruntime/rolloutKind:RolloutKind default {{project}}/{{location}}/{{rollout_kind_id}}
 * $ pulumi import gcp:saasruntime/rolloutKind:RolloutKind default {{location}}/{{rollout_kind_id}}
 * ```
 */
class RolloutKind extends pulumi.CustomResource {
    /**
     * Get an existing RolloutKind resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RolloutKind(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RolloutKind.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RolloutKind.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["annotations"] = state?.annotations;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["effectiveAnnotations"] = state?.effectiveAnnotations;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["errorBudget"] = state?.errorBudget;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["rolloutKindId"] = state?.rolloutKindId;
            resourceInputs["rolloutOrchestrationStrategy"] = state?.rolloutOrchestrationStrategy;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["unitFilter"] = state?.unitFilter;
            resourceInputs["unitKind"] = state?.unitKind;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["updateUnitKindStrategy"] = state?.updateUnitKindStrategy;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.rolloutKindId === undefined && !opts.urn) {
                throw new Error("Missing required property 'rolloutKindId'");
            }
            if (args?.unitKind === undefined && !opts.urn) {
                throw new Error("Missing required property 'unitKind'");
            }
            resourceInputs["annotations"] = args?.annotations;
            resourceInputs["errorBudget"] = args?.errorBudget;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["rolloutKindId"] = args?.rolloutKindId;
            resourceInputs["rolloutOrchestrationStrategy"] = args?.rolloutOrchestrationStrategy;
            resourceInputs["unitFilter"] = args?.unitFilter;
            resourceInputs["unitKind"] = args?.unitKind;
            resourceInputs["updateUnitKindStrategy"] = args?.updateUnitKindStrategy;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveAnnotations"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(RolloutKind.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RolloutKind = RolloutKind;
/** @internal */
RolloutKind.__pulumiType = 'gcp:saasruntime/rolloutKind:RolloutKind';
