// @pulumi/gcp/privilegedaccessmanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Settings = exports.getEntitlementOutput = exports.getEntitlement = exports.Entitlement = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Entitlement = null;
utilities.lazyLoad(exports, ["Entitlement"], () => require("./entitlement"));
exports.getEntitlement = null;
exports.getEntitlementOutput = null;
utilities.lazyLoad(exports, ["getEntitlement", "getEntitlementOutput"], () => require("./getEntitlement"));
exports.Settings = null;
utilities.lazyLoad(exports, ["Settings"], () => require("./settings"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:privilegedaccessmanager/entitlement:entitlement":
                return new exports.Entitlement(name, undefined, { urn });
            case "gcp:privilegedaccessmanager/settings:Settings":
                return new exports.Settings(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "privilegedaccessmanager/entitlement", _module);
pulumi.runtime.registerResourceModule("gcp", "privilegedaccessmanager/settings", _module);
