// @pulumi/gcp/securitycenter/managementOrganizationEventThreatDetectionCustomModule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementOrganizationEventThreatDetectionCustomModule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents an instance of an Event Threat Detection custom module, including
 * its full module name, display name, enablement state, and last updated time.
 * You can create a custom module at the organization level only.
 *
 * To get more information about OrganizationEventThreatDetectionCustomModule, see:
 *
 * * [API documentation](https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules)
 * * How-to Guides
 *     * [Overview of custom modules for Event Threat Detection](https://cloud.google.com/security-command-center/docs/custom-modules-etd-overview)
 *
 * ## Example Usage
 *
 * ### Scc Management Organization Event Threat Detection Custom Module
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.securitycenter.ManagementOrganizationEventThreatDetectionCustomModule("example", {
 *     organization: "123456789",
 *     location: "global",
 *     displayName: "basic_custom_module",
 *     enablementState: "ENABLED",
 *     type: "CONFIGURABLE_BAD_IP",
 *     description: "My Event Threat Detection Custom Module",
 *     config: JSON.stringify({
 *         metadata: {
 *             severity: "LOW",
 *             description: "Flagged by Forcepoint as malicious",
 *             recommendation: "Contact the owner of the relevant project.",
 *         },
 *         ips: [
 *             "192.0.2.1",
 *             "192.0.2.0/24",
 *         ],
 *     }),
 * });
 * ```
 *
 * ## Import
 *
 * OrganizationEventThreatDetectionCustomModule can be imported using any of these accepted formats:
 *
 * * `organizations/{{organization}}/locations/{{location}}/eventThreatDetectionCustomModules/{{name}}`
 * * `{{organization}}/{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, OrganizationEventThreatDetectionCustomModule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:securitycenter/managementOrganizationEventThreatDetectionCustomModule:ManagementOrganizationEventThreatDetectionCustomModule default organizations/{{organization}}/locations/{{location}}/eventThreatDetectionCustomModules/{{name}}
 * $ pulumi import gcp:securitycenter/managementOrganizationEventThreatDetectionCustomModule:ManagementOrganizationEventThreatDetectionCustomModule default {{organization}}/{{location}}/{{name}}
 * ```
 */
class ManagementOrganizationEventThreatDetectionCustomModule extends pulumi.CustomResource {
    /**
     * Get an existing ManagementOrganizationEventThreatDetectionCustomModule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ManagementOrganizationEventThreatDetectionCustomModule(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ManagementOrganizationEventThreatDetectionCustomModule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementOrganizationEventThreatDetectionCustomModule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state?.config;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["enablementState"] = state?.enablementState;
            resourceInputs["lastEditor"] = state?.lastEditor;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["organization"] = state?.organization;
            resourceInputs["type"] = state?.type;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.organization === undefined && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            resourceInputs["config"] = args?.config;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["enablementState"] = args?.enablementState;
            resourceInputs["location"] = args?.location;
            resourceInputs["organization"] = args?.organization;
            resourceInputs["type"] = args?.type;
            resourceInputs["lastEditor"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ManagementOrganizationEventThreatDetectionCustomModule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagementOrganizationEventThreatDetectionCustomModule = ManagementOrganizationEventThreatDetectionCustomModule;
/** @internal */
ManagementOrganizationEventThreatDetectionCustomModule.__pulumiType = 'gcp:securitycenter/managementOrganizationEventThreatDetectionCustomModule:ManagementOrganizationEventThreatDetectionCustomModule';
