// @pulumi/gcp/developerconnect/insightsConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsightsConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Description
 *
 * ## Example Usage
 *
 * ### Developer Connect Insights Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 * import * as time from "@pulumiverse/time";
 *
 * const project = new gcp.organizations.Project("project", {
 *     projectId: "dci-tf-_10719",
 *     name: "Service Project",
 *     orgId: "123456789",
 *     billingAccount: "000000-0000000-0000000-000000",
 *     deletionPolicy: "DELETE",
 * });
 * // Grant Permissions
 * const apphubPermissions = new gcp.projects.IAMMember("apphub_permissions", {
 *     project: project.projectId,
 *     role: "roles/apphub.admin",
 *     member: "serviceAccount:hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com",
 * });
 * const insightsAgent = new gcp.projects.IAMMember("insights_agent", {
 *     project: project.projectId,
 *     role: "roles/developerconnect.insightsAgent",
 *     member: "serviceAccount:66214305248-compute@developer.gserviceaccount.com",
 * });
 * // Enable APIs
 * const apphubApiService = new gcp.projects.Service("apphub_api_service", {
 *     project: project.projectId,
 *     service: "apphub.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containeranalysisApi = new gcp.projects.Service("containeranalysis_api", {
 *     project: project.projectId,
 *     service: "containeranalysis.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containerscanningApi = new gcp.projects.Service("containerscanning_api", {
 *     project: project.projectId,
 *     service: "containerscanning.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containerApi = new gcp.projects.Service("container_api", {
 *     project: project.projectId,
 *     service: "container.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const artifactregistryApi = new gcp.projects.Service("artifactregistry_api", {
 *     project: project.projectId,
 *     service: "artifactregistry.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const cloudbuildApi = new gcp.projects.Service("cloudbuild_api", {
 *     project: project.projectId,
 *     service: "cloudbuild.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const cloudassetApi = new gcp.projects.Service("cloudasset_api", {
 *     project: project.projectId,
 *     service: "cloudasset.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const computeApi = new gcp.projects.Service("compute_api", {
 *     project: project.projectId,
 *     service: "compute.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const devconnectApi = new gcp.projects.Service("devconnect_api", {
 *     project: project.projectId,
 *     service: "developerconnect.googleapis.com",
 * }, {
 *     dependsOn: [project],
 * });
 * // Wait delay after enabling APIs and granting permissions
 * const waitForPropagation = new time.Sleep("wait_for_propagation", {createDuration: "120s"}, {
 *     dependsOn: [
 *         apphubPermissions,
 *         insightsAgent,
 *         apphubApiService,
 *         containeranalysisApi,
 *         containerscanningApi,
 *         containerApi,
 *         artifactregistryApi,
 *         artifactregistryApi,
 *         cloudbuildApi,
 *         cloudassetApi,
 *         computeApi,
 *         devconnectApi,
 *     ],
 * });
 * const myApphubApplication = new gcp.apphub.Application("my_apphub_application", {
 *     location: "us-central1",
 *     applicationId: "tf-test-example-application_1443",
 *     scope: {
 *         type: "REGIONAL",
 *     },
 *     project: project.projectId,
 * }, {
 *     dependsOn: [waitForPropagation],
 * });
 * const insightsConfig = new gcp.developerconnect.InsightsConfig("insights_config", {
 *     location: "us-central1",
 *     insightsConfigId: "tf-test-ic-apphub-_26032",
 *     project: project.projectId,
 *     annotations: {},
 *     labels: {},
 *     appHubApplication: std.format({
 *         input: "//apphub.googleapis.com/projects/%s/locations/%s/applications/%s",
 *         args: [
 *             project.number,
 *             myApphubApplication.location,
 *             myApphubApplication.applicationId,
 *         ],
 *     }).then(invoke => invoke.result),
 *     artifactConfigs: [{
 *         googleArtifactAnalysis: {
 *             projectId: project.projectId,
 *         },
 *         googleArtifactRegistry: {
 *             artifactRegistryPackage: "my-package",
 *             projectId: project.projectId,
 *         },
 *         uri: "us-docker.pkg.dev/my-project/my-repo/my-image",
 *     }],
 * }, {
 *     dependsOn: [waitForPropagation],
 * });
 * ```
 * ### Developer Connect Insights Config Projects
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumiverse/time";
 *
 * const project = new gcp.organizations.Project("project", {
 *     projectId: "dci-tf-_8647",
 *     name: "Service Project",
 *     orgId: "123456789",
 *     billingAccount: "000000-0000000-0000000-000000",
 *     deletionPolicy: "DELETE",
 * });
 * // Grant Permissions
 * const apphubPermissions = new gcp.projects.IAMMember("apphub_permissions", {
 *     project: project.projectId,
 *     role: "roles/apphub.admin",
 *     member: "serviceAccount:hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com",
 * });
 * const insightsAgent = new gcp.projects.IAMMember("insights_agent", {
 *     project: project.projectId,
 *     role: "roles/developerconnect.insightsAgent",
 *     member: "serviceAccount:66214305248-compute@developer.gserviceaccount.com",
 * });
 * // Enable APIs
 * const apphubApiService = new gcp.projects.Service("apphub_api_service", {
 *     project: project.projectId,
 *     service: "apphub.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containeranalysisApi = new gcp.projects.Service("containeranalysis_api", {
 *     project: project.projectId,
 *     service: "containeranalysis.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containerscanningApi = new gcp.projects.Service("containerscanning_api", {
 *     project: project.projectId,
 *     service: "containerscanning.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const containerApi = new gcp.projects.Service("container_api", {
 *     project: project.projectId,
 *     service: "container.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const artifactregistryApi = new gcp.projects.Service("artifactregistry_api", {
 *     project: project.projectId,
 *     service: "artifactregistry.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const cloudbuildApi = new gcp.projects.Service("cloudbuild_api", {
 *     project: project.projectId,
 *     service: "cloudbuild.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const cloudassetApi = new gcp.projects.Service("cloudasset_api", {
 *     project: project.projectId,
 *     service: "cloudasset.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const computeApi = new gcp.projects.Service("compute_api", {
 *     project: project.projectId,
 *     service: "compute.googleapis.com",
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [project],
 * });
 * const devconnectApi = new gcp.projects.Service("devconnect_api", {
 *     project: project.projectId,
 *     service: "developerconnect.googleapis.com",
 * }, {
 *     dependsOn: [project],
 * });
 * // Wait delay after enabling APIs and granting permissions
 * const waitForPropagation = new time.Sleep("wait_for_propagation", {createDuration: "120s"}, {
 *     dependsOn: [
 *         apphubPermissions,
 *         insightsAgent,
 *         apphubApiService,
 *         containeranalysisApi,
 *         containerscanningApi,
 *         containerApi,
 *         artifactregistryApi,
 *         artifactregistryApi,
 *         cloudbuildApi,
 *         cloudassetApi,
 *         computeApi,
 *         devconnectApi,
 *     ],
 * });
 * const insightsConfigProjects = new gcp.developerconnect.InsightsConfig("insights_config_projects", {
 *     location: "us-central1",
 *     insightsConfigId: "tf-test-ic-projects-_50610",
 *     project: project.projectId,
 *     annotations: {},
 *     labels: {},
 *     targetProjects: {
 *         projectIds: [pulumi.interpolate`projects/${project.projectId}`],
 *     },
 *     artifactConfigs: [{
 *         googleArtifactAnalysis: {
 *             projectId: project.projectId,
 *         },
 *         googleArtifactRegistry: {
 *             artifactRegistryPackage: "my-package",
 *             projectId: project.projectId,
 *         },
 *         uri: "us-docker.pkg.dev/my-project/my-repo/my-image",
 *     }],
 * }, {
 *     dependsOn: [waitForPropagation],
 * });
 * ```
 *
 * ## Import
 *
 * InsightsConfig can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/insightsConfigs/{{insights_config_id}}`
 * * `{{project}}/{{location}}/{{insights_config_id}}`
 * * `{{location}}/{{insights_config_id}}`
 *
 * When using the `pulumi import` command, InsightsConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:developerconnect/insightsConfig:InsightsConfig default projects/{{project}}/locations/{{location}}/insightsConfigs/{{insights_config_id}}
 * $ pulumi import gcp:developerconnect/insightsConfig:InsightsConfig default {{project}}/{{location}}/{{insights_config_id}}
 * $ pulumi import gcp:developerconnect/insightsConfig:InsightsConfig default {{location}}/{{insights_config_id}}
 * ```
 */
class InsightsConfig extends pulumi.CustomResource {
    /**
     * Get an existing InsightsConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new InsightsConfig(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of InsightsConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InsightsConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["annotations"] = state?.annotations;
            resourceInputs["appHubApplication"] = state?.appHubApplication;
            resourceInputs["artifactConfigs"] = state?.artifactConfigs;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["effectiveAnnotations"] = state?.effectiveAnnotations;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["errors"] = state?.errors;
            resourceInputs["insightsConfigId"] = state?.insightsConfigId;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["reconciling"] = state?.reconciling;
            resourceInputs["runtimeConfigs"] = state?.runtimeConfigs;
            resourceInputs["state"] = state?.state;
            resourceInputs["targetProjects"] = state?.targetProjects;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.insightsConfigId === undefined && !opts.urn) {
                throw new Error("Missing required property 'insightsConfigId'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["annotations"] = args?.annotations;
            resourceInputs["appHubApplication"] = args?.appHubApplication;
            resourceInputs["artifactConfigs"] = args?.artifactConfigs;
            resourceInputs["insightsConfigId"] = args?.insightsConfigId;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["targetProjects"] = args?.targetProjects;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveAnnotations"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["reconciling"] = undefined /*out*/;
            resourceInputs["runtimeConfigs"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(InsightsConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InsightsConfig = InsightsConfig;
/** @internal */
InsightsConfig.__pulumiType = 'gcp:developerconnect/insightsConfig:InsightsConfig';
