// @pulumi/gcp/cloudidentity/getPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPolicyOutput = exports.getPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Use this data source to retrieve a Cloud Identity policy.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const test = gcp.cloudidentity.getPolicy({
 *     name: "policies/{policy_id}",
 * });
 * export const policyCustomer = test.then(test => test.customer);
 * export const policyQueryQuery = test.then(test => test.policyQueries?.[0]?.query);
 * export const policyQueryOrgUnit = test.then(test => test.policyQueries?.[0]?.orgUnit);
 * export const policyQueryGroup = test.then(test => test.policyQueries?.[0]?.group);
 * export const policyQuerySortOrder = test.then(test => test.policyQueries?.[0]?.sortOrder);
 * export const policySetting = test.then(test => test.setting);
 * ```
 */
function getPolicy(args, opts) {
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invoke("gcp:cloudidentity/getPolicy:getPolicy", {
        "name": args.name,
    }, opts);
}
exports.getPolicy = getPolicy;
/**
 * Use this data source to retrieve a Cloud Identity policy.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const test = gcp.cloudidentity.getPolicy({
 *     name: "policies/{policy_id}",
 * });
 * export const policyCustomer = test.then(test => test.customer);
 * export const policyQueryQuery = test.then(test => test.policyQueries?.[0]?.query);
 * export const policyQueryOrgUnit = test.then(test => test.policyQueries?.[0]?.orgUnit);
 * export const policyQueryGroup = test.then(test => test.policyQueries?.[0]?.group);
 * export const policyQuerySortOrder = test.then(test => test.policyQueries?.[0]?.sortOrder);
 * export const policySetting = test.then(test => test.setting);
 * ```
 */
function getPolicyOutput(args, opts) {
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invokeOutput("gcp:cloudidentity/getPolicy:getPolicy", {
        "name": args.name,
    }, opts);
}
exports.getPolicyOutput = getPolicyOutput;
