// @pulumi/gcp/cloudidentity/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = exports.GroupMembership = exports.Group = exports.getPolicyOutput = exports.getPolicy = exports.getPoliciesOutput = exports.getPolicies = exports.getGroupsOutput = exports.getGroups = exports.getGroupTransitiveMembershipsOutput = exports.getGroupTransitiveMemberships = exports.getGroupMembershipsOutput = exports.getGroupMemberships = exports.getGroupLookupOutput = exports.getGroupLookup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroupLookup = null;
exports.getGroupLookupOutput = null;
utilities.lazyLoad(exports, ["getGroupLookup", "getGroupLookupOutput"], () => require("./getGroupLookup"));
exports.getGroupMemberships = null;
exports.getGroupMembershipsOutput = null;
utilities.lazyLoad(exports, ["getGroupMemberships", "getGroupMembershipsOutput"], () => require("./getGroupMemberships"));
exports.getGroupTransitiveMemberships = null;
exports.getGroupTransitiveMembershipsOutput = null;
utilities.lazyLoad(exports, ["getGroupTransitiveMemberships", "getGroupTransitiveMembershipsOutput"], () => require("./getGroupTransitiveMemberships"));
exports.getGroups = null;
exports.getGroupsOutput = null;
utilities.lazyLoad(exports, ["getGroups", "getGroupsOutput"], () => require("./getGroups"));
exports.getPolicies = null;
exports.getPoliciesOutput = null;
utilities.lazyLoad(exports, ["getPolicies", "getPoliciesOutput"], () => require("./getPolicies"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupMembership = null;
utilities.lazyLoad(exports, ["GroupMembership"], () => require("./groupMembership"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudidentity/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "gcp:cloudidentity/groupMembership:GroupMembership":
                return new exports.GroupMembership(name, undefined, { urn });
            case "gcp:cloudidentity/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudidentity/group", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudidentity/groupMembership", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudidentity/policy", _module);
