// @pulumi/gcp/cloudidentity/policy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Cloud Identity Policy binds a Setting to a PolicyQuery for a Google Workspace / Cloud Identity customer.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about Policy, see:
 *
 * * [API documentation](https://cloud.google.com/identity/docs/reference/rest/v1beta1/policies)
 * * How-to Guides
 *     * [Policy API overview](https://docs.cloud.google.com/identity/docs/concepts/overview-policies)
 *
 * > **Note:** This is available only in beta
 * ## Example Usage
 *
 * ### Cloudidentity Policy Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const primary = new gcp.cloudidentity.Policy("primary", {
 *     customer: "customers/C01234567",
 *     policyQuery: {
 *         orgUnit: "orgUnits/03abcxyz",
 *         group: "groups/0123456789",
 *         query: "true",
 *     },
 *     setting: {
 *         type: "something.googleapis.com/SettingType",
 *         valueJson: JSON.stringify({
 *             enabled: true,
 *         }),
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Policy can be imported using any of these accepted formats:
 *
 * * `policies/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, Policy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:cloudidentity/policy:Policy default policies/{{name}}
 * $ pulumi import gcp:cloudidentity/policy:Policy default {{name}}
 * ```
 */
class Policy extends pulumi.CustomResource {
    /**
     * Get an existing Policy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Policy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Policy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Policy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customer"] = state?.customer;
            resourceInputs["name"] = state?.name;
            resourceInputs["policyQuery"] = state?.policyQuery;
            resourceInputs["setting"] = state?.setting;
        }
        else {
            const args = argsOrState;
            if (args?.customer === undefined && !opts.urn) {
                throw new Error("Missing required property 'customer'");
            }
            if (args?.policyQuery === undefined && !opts.urn) {
                throw new Error("Missing required property 'policyQuery'");
            }
            if (args?.setting === undefined && !opts.urn) {
                throw new Error("Missing required property 'setting'");
            }
            resourceInputs["customer"] = args?.customer;
            resourceInputs["policyQuery"] = args?.policyQuery;
            resourceInputs["setting"] = args?.setting;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Policy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Policy = Policy;
/** @internal */
Policy.__pulumiType = 'gcp:cloudidentity/policy:Policy';
