// @pulumi/gcp/dataplex/dataAsset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataAsset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A data asset resource that can be packaged and shared via a data product.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about DataAsset, see:
 *
 * * [API documentation](https://cloud.google.com/dataplex/docs/reference/rest/v1/projects.locations.dataProducts.dataAssets)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/dataplex/docs)
 *
 * ## Example Usage
 *
 * ### Dataplex Data Asset Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.dataplex.DataProduct("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: "tf-test-dp-_64336",
 *     displayName: "Parent Data Product",
 *     ownerEmails: ["gterraformtestuser@gmail.com"],
 *     accessGroups: [{
 *         id: "analyst",
 *         groupId: "analyst",
 *         displayName: "Data Analyst",
 *         principal: {
 *             googleGroup: "tf-test-analysts-_34962@example.com",
 *         },
 *     }],
 * });
 * const exampleDataset = new gcp.bigquery.Dataset("example", {
 *     project: "my-project-name",
 *     datasetId: "tf_test_dataset__74000",
 *     location: "us-central1",
 * });
 * const exampleDataAsset = new gcp.dataplex.DataAsset("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: example.dataProductId,
 *     dataAssetId: "data-asset",
 *     resource: pulumi.interpolate`//bigquery.googleapis.com/projects/${exampleDataset.project}/datasets/${exampleDataset.datasetId}`,
 * });
 * ```
 * ### Dataplex Data Asset Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.dataplex.DataProduct("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: "tf-test-dp-_75125",
 *     displayName: "Full Example Parent DP",
 *     ownerEmails: ["gterraformtestuser@gmail.com"],
 *     accessGroups: [
 *         {
 *             id: "analyst",
 *             groupId: "analyst",
 *             displayName: "Data Analyst",
 *             principal: {
 *                 googleGroup: "dataproduct-terraform-examples@google.com",
 *             },
 *         },
 *         {
 *             id: "scientist",
 *             groupId: "scientist",
 *             displayName: "Data Scientist",
 *             principal: {
 *                 googleGroup: "dataproduct-terraform-examples-2@google.com",
 *             },
 *         },
 *     ],
 * });
 * const exampleDataset = new gcp.bigquery.Dataset("example", {
 *     project: "my-project-name",
 *     datasetId: "tf_test_dataset__88722",
 *     location: "us-central1",
 * });
 * const exampleDataAsset = new gcp.dataplex.DataAsset("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: example.dataProductId,
 *     dataAssetId: "data-asset",
 *     resource: pulumi.interpolate`//bigquery.googleapis.com/projects/${exampleDataset.project}/datasets/${exampleDataset.datasetId}`,
 *     labels: {
 *         env: "prod",
 *         critical: "true",
 *     },
 *     accessGroupConfigs: [
 *         {
 *             accessGroup: "analyst",
 *             iamRoles: ["roles/bigquery.dataViewer"],
 *         },
 *         {
 *             accessGroup: "scientist",
 *             iamRoles: ["roles/bigquery.dataEditor"],
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * DataAsset can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/dataProducts/{{data_product_id}}/dataAssets/{{data_asset_id}}`
 * * `{{project}}/{{location}}/{{data_product_id}}/{{data_asset_id}}`
 * * `{{location}}/{{data_product_id}}/{{data_asset_id}}`
 *
 * When using the `pulumi import` command, DataAsset can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataplex/dataAsset:DataAsset default projects/{{project}}/locations/{{location}}/dataProducts/{{data_product_id}}/dataAssets/{{data_asset_id}}
 * $ pulumi import gcp:dataplex/dataAsset:DataAsset default {{project}}/{{location}}/{{data_product_id}}/{{data_asset_id}}
 * $ pulumi import gcp:dataplex/dataAsset:DataAsset default {{location}}/{{data_product_id}}/{{data_asset_id}}
 * ```
 */
class DataAsset extends pulumi.CustomResource {
    /**
     * Get an existing DataAsset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DataAsset(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DataAsset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataAsset.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessGroupConfigs"] = state?.accessGroupConfigs;
            resourceInputs["dataAssetId"] = state?.dataAssetId;
            resourceInputs["dataProductId"] = state?.dataProductId;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["resource"] = state?.resource;
            resourceInputs["uid"] = state?.uid;
        }
        else {
            const args = argsOrState;
            if (args?.dataAssetId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataAssetId'");
            }
            if (args?.dataProductId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataProductId'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.resource === undefined && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            resourceInputs["accessGroupConfigs"] = args?.accessGroupConfigs;
            resourceInputs["dataAssetId"] = args?.dataAssetId;
            resourceInputs["dataProductId"] = args?.dataProductId;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["resource"] = args?.resource;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(DataAsset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataAsset = DataAsset;
/** @internal */
DataAsset.__pulumiType = 'gcp:dataplex/dataAsset:DataAsset';
