// @pulumi/gcp/gkehub/rolloutSequence.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RolloutSequence = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * RolloutSequence defines the desired order of upgrades.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about RolloutSequence, see:
 *
 * * [API documentation](https://docs.cloud.google.com/kubernetes-engine/fleet-management/docs/reference/rest/v1beta/projects.locations.rolloutSequences)
 * * How-to Guides
 *     * [Rollout Sequencing Overview](https://cloud.google.com/kubernetes-engine/docs/concepts/rollout-sequencing-custom-stages/about-rollout-sequencing)
 *
 * ## Import
 *
 * RolloutSequence can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/global/rolloutSequences/{{rollout_sequence_id}}`
 * * `{{project}}/{{rollout_sequence_id}}`
 * * `{{rollout_sequence_id}}`
 *
 * When using the `pulumi import` command, RolloutSequence can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:gkehub/rolloutSequence:RolloutSequence default projects/{{project}}/locations/global/rolloutSequences/{{rollout_sequence_id}}
 * $ pulumi import gcp:gkehub/rolloutSequence:RolloutSequence default {{project}}/{{rollout_sequence_id}}
 * $ pulumi import gcp:gkehub/rolloutSequence:RolloutSequence default {{rollout_sequence_id}}
 * ```
 */
class RolloutSequence extends pulumi.CustomResource {
    /**
     * Get an existing RolloutSequence resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RolloutSequence(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RolloutSequence.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RolloutSequence.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deleteTime"] = state?.deleteTime;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["rolloutSequenceId"] = state?.rolloutSequenceId;
            resourceInputs["stages"] = state?.stages;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.rolloutSequenceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'rolloutSequenceId'");
            }
            if (args?.stages === undefined && !opts.urn) {
                throw new Error("Missing required property 'stages'");
            }
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["project"] = args?.project;
            resourceInputs["rolloutSequenceId"] = args?.rolloutSequenceId;
            resourceInputs["stages"] = args?.stages;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["deleteTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(RolloutSequence.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RolloutSequence = RolloutSequence;
/** @internal */
RolloutSequence.__pulumiType = 'gcp:gkehub/rolloutSequence:RolloutSequence';
